docs/HyperIndex/overview.md

```diff
@@ -49,6 +49,20 @@ While HyperIndex gives you a complete indexing solution with schema management a
 
 ---
 
+## Hypersync API Token Requirements
+
+Starting from **9 May 2024**, HyperSync (the data engine powering HyperIndex) will implement rate limits for requests without API tokens. Here's what you need to know:
+
+- **Local Development**: No API token is required for local development, though requests will be rate limited
+- **Self-Hosted Deployments**: API tokens are required for unlimited HyperSync access in self-hosted deployments. The token can be set via the `ENVIO_API_TOKEN` environment variable in your indexer configuration
```

Comment:
```suggestion
- **Self-Hosted Deployments**: API tokens are required for unlimited HyperSync access in self-hosted deployments. The token can be set via the `ENVIO_API_TOKEN` environment variable in your indexer configuration. This can be read from the `.env` file in the root of your HyperIndex project.
```